cmd/server-main.go

```diff
@@ -972,11 +972,37 @@ func serverMain(ctx *cli.Context) {
 	}
 
 	// Initialize users credentials and policies in background right after config has initialized.
+	globalIAMFullyInitialized.Store(true)
 	go func() {
 		bootstrapTrace("globalIAMSys.Init", func() {
 			globalIAMSys.Init(GlobalContext, newObject, globalEtcdClient, globalRefreshIAMInterval)
 		})
 
+		// Start a thread to keep checking IAM initialized and re-trigger
+		go func() {
+			ticker := time.NewTicker(10 * time.Second) // Check every 10 seconds
+			defer ticker.Stop()
+			for {
+				select {
+				case <-ticker.C:
+					if !globalIAMFullyInitialized.Load() {
+						globalIAMSys.Init(GlobalContext, newObject, globalEtcdClient, globalRefreshIAMInterval)
```

Comment:
Without checking the details, it seems like `go sys.periodicRoutines(ctx, refreshInterval)` will be called each time you call Init.  Maybe it should only be called if successful?

Reply:
yes, I will change accordingly and verify changes.